<Type Name="AccessViolationException" FullName="System.AccessViolationException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="eb360adfc3670886a06c5f3ab2daa2470318896a" />
    <Meta Name="ms.sourcegitcommit" Value="d31dc2ede16f6f7bc64e90d9f897ff54c4e3869b" />
    <Meta Name="ms.translationtype" Value="MT" />
    <Meta Name="ms.contentlocale" Value="pl-PL" />
    <Meta Name="ms.lasthandoff" Value="04/03/2018" />
    <Meta Name="ms.locfileid" Value="30743789" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class AccessViolationException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit AccessViolationException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.AccessViolationException" />
  <TypeSignature Language="VB.NET" Value="Public Class AccessViolationException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class AccessViolationException : SystemException" />
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <span data-ttu-id="69f1e-101">Wyjątek zgłaszany, gdy jest próba odczytu lub zapisu chronione pamięci.</span>
      <span class="sxs-lookup">
        <span data-stu-id="69f1e-101">The exception that is thrown when there is an attempt to read or write protected memory.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="69f1e-102">Naruszenie zasad dostępu występuje w kodu niezarządzanego lub niebezpieczny, gdy kod podejmuje próbę odczytu lub zapisu w pamięci, która nie została przydzielona, lub do których nie ma dostępu.</span><span class="sxs-lookup"><span data-stu-id="69f1e-102">An access violation occurs in unmanaged or unsafe code when the code attempts to read or write to memory that has not been allocated, or to which it does not have access.</span></span> <span data-ttu-id="69f1e-103">Najczęstszą przyczyną wskaźnik ma nieprawidłowe wartości.</span><span class="sxs-lookup"><span data-stu-id="69f1e-103">This usually occurs because a pointer has a bad value.</span></span> <span data-ttu-id="69f1e-104">Nie wszystkie zapisuje lub odczytuje za pośrednictwem nieprawidłowe wskaźniki prowadzić do naruszenia zasad dostępu, więc naruszenia zasad dostępu zwykle wskazuje, że wystąpiły kilka odczytów i zapisów przez nieprawidłowe wskaźniki i pamięci, może być uszkodzony.</span><span class="sxs-lookup"><span data-stu-id="69f1e-104">Not all reads or writes through bad pointers lead to access violations, so an access violation usually indicates that several reads or writes have occurred through bad pointers, and that memory might be corrupted.</span></span> <span data-ttu-id="69f1e-105">W związku z tym naruszenia zasad dostępu prawie zawsze wskazują poważne błędy programowania.</span><span class="sxs-lookup"><span data-stu-id="69f1e-105">Thus, access violations almost always indicate serious programming errors.</span></span> <span data-ttu-id="69f1e-106"><xref:System.AccessViolationException> Jasno identyfikuje tych błędów.</span><span class="sxs-lookup"><span data-stu-id="69f1e-106">An <xref:System.AccessViolationException> clearly identifies these serious errors.</span></span>  
  
 <span data-ttu-id="69f1e-107">W programach składające się wyłącznie z weryfikowalny kod zarządzany wszystkie odwołania jest nieprawidłowy lub ma wartość null, a naruszenia zasad dostępu jest niemożliwe.</span><span class="sxs-lookup"><span data-stu-id="69f1e-107">In programs consisting entirely of verifiable managed code, all references are either valid or null, and access violations are impossible.</span></span> <span data-ttu-id="69f1e-108">Wszelkie operacje, który próbuje odwołać odwołanie o wartości null w zgłasza weryfikowalny kod <xref:System.NullReferenceException> wyjątku.</span><span class="sxs-lookup"><span data-stu-id="69f1e-108">Any operation that attempts to reference a null reference in verifiable code throws a <xref:System.NullReferenceException> exception.</span></span> <span data-ttu-id="69f1e-109"><xref:System.AccessViolationException> Występuje tylko gdy weryfikowalny kod zarządzany wchodzi w interakcję z kodem niezarządzanym lub niebezpiecznego kodu zarządzanego.</span><span class="sxs-lookup"><span data-stu-id="69f1e-109">An <xref:System.AccessViolationException> occurs only when verifiable managed code interacts with unmanaged code or with unsafe managed code.</span></span>  
  
## <a name="troubleshooting-accessviolationexception-exceptions"></a><span data-ttu-id="69f1e-110">Rozwiązywanie problemów z wyjątkami accessviolationexception —</span><span class="sxs-lookup"><span data-stu-id="69f1e-110">Troubleshooting AccessViolationException exceptions</span></span>  
 <span data-ttu-id="69f1e-111"><xref:System.AccessViolationException> Wyjątek może wystąpić tylko w niebezpiecznego kodu zarządzanego lub gdy zarządzany weryfikowalny kod współdziała z kodem niezarządzanym:</span><span class="sxs-lookup"><span data-stu-id="69f1e-111">An <xref:System.AccessViolationException> exception can occur only in unsafe managed code or when verifiable managed code interacts with unmanaged code:</span></span>  
  
-   <span data-ttu-id="69f1e-112">Naruszenia zasad dostępu występujący w niebezpieczny kod zarządzany może zostać wyrażona jako <xref:System.NullReferenceException> wyjątek lub <xref:System.AccessViolationException> wyjątku, w zależności od platformy.</span><span class="sxs-lookup"><span data-stu-id="69f1e-112">An access violation that occurs in unsafe managed code can be expressed as either a <xref:System.NullReferenceException> exception or an <xref:System.AccessViolationException> exception, depending on the platform.</span></span>  
  
-   <span data-ttu-id="69f1e-113">Naruszenia zasad dostępu za pomocą kodu niezarządzanego, które propaguje do kodu zarządzanego zawsze jest ujęte w <xref:System.AccessViolationException> wyjątku.</span><span class="sxs-lookup"><span data-stu-id="69f1e-113">An access violation in unmanaged code that bubbles up to managed code is always wrapped in an <xref:System.AccessViolationException> exception.</span></span>  
  
 <span data-ttu-id="69f1e-114">W obu przypadkach można zidentyfikować i rozwiązać przyczynę <xref:System.AccessViolationException> wyjątek w następujący sposób:</span><span class="sxs-lookup"><span data-stu-id="69f1e-114">In either case, you can identify and correct the cause of the <xref:System.AccessViolationException> exception as follows:</span></span>  
  
 <span data-ttu-id="69f1e-115">Upewnij się, przydzielono pamięci, który próbujesz uzyskać dostęp.</span><span class="sxs-lookup"><span data-stu-id="69f1e-115">Make sure that the memory that you are attempting to access has been allocated.</span></span>  
 <span data-ttu-id="69f1e-116"><xref:System.AccessViolationException> Jest zawsze wyjątek przy próbie uzyskania dostępu do chronionej pamięci — to znaczy, aby uzyskiwać dostęp do pamięci nie jest przydzielony ani nie posiada przez proces.</span><span class="sxs-lookup"><span data-stu-id="69f1e-116">An <xref:System.AccessViolationException> exception is always thrown by an attempt to access protected memory -- that is, to access memory that is not allocated or that is not owned by a process.</span></span>  
  
 <span data-ttu-id="69f1e-117">Automatyczne zarządzanie pamięcią jest jednym z usług, które udostępnia środowisko uruchomieniowe języka wspólnego.</span><span class="sxs-lookup"><span data-stu-id="69f1e-117">Automatic memory management is one of the services that the common language runtime provides.</span></span> <span data-ttu-id="69f1e-118">Kod zarządzany zapewnia te same funkcje co kodu unmanagede, warto przejdź do kodu zarządzanego, aby móc korzystać z tej funkcji.</span><span class="sxs-lookup"><span data-stu-id="69f1e-118">If managed code provides the same functionality as your unmanagede code, you may wish to move to managed code to take advantage of this functionality.</span></span> <span data-ttu-id="69f1e-119">Aby uzyskać więcej informacji, zobacz [automatyczne zarządzanie pamięcią](~/docs/standard/automatic-memory-management.md).</span><span class="sxs-lookup"><span data-stu-id="69f1e-119">For more information, see [Automatic Memory Management](~/docs/standard/automatic-memory-management.md).</span></span>  
  
 <span data-ttu-id="69f1e-120">Upewnij się, że są próby uzyskania dostępu do pamięci nie jest uszkodzony.</span><span class="sxs-lookup"><span data-stu-id="69f1e-120">Make sure that the memory that you are attempting to access has not been corrupted.</span></span>  
 <span data-ttu-id="69f1e-121">Jeśli kilka odczytu lub zapisu wystąpiły za pośrednictwem nieprawidłowe wskaźniki, pamięci może być uszkodzony.</span><span class="sxs-lookup"><span data-stu-id="69f1e-121">If several read or write operations have occurred through bad pointers, memory may be corrupted.</span></span> <span data-ttu-id="69f1e-122">Ten błąd zazwyczaj występuje podczas odczytywania lub zapisywania adresów poza wstępnie zdefiniowanych buforu.</span><span class="sxs-lookup"><span data-stu-id="69f1e-122">This typically occurs when reading or writing to addresses outside of a predefined buffer.</span></span>  
  
## <a name="accessviolationexception-and-trycatch-blocks"></a><span data-ttu-id="69f1e-123">Bloki accessviolationexception — i try/catch</span><span class="sxs-lookup"><span data-stu-id="69f1e-123">AccessViolationException and try/catch blocks</span></span>  
 <span data-ttu-id="69f1e-124">Począwszy od [!INCLUDE[net_v40_long](~/includes/net-v40-long-md.md)], <xref:System.AccessViolationException> wyjątków zgłaszanych przez środowisko uruchomieniowe języka wspólnego nie są obsługiwane przez `catch` instrukcji programu obsługi wyjątków strukturalnych, jeśli wystąpi wyjątek poza pamięć zarezerwowana przez języka wspólnego środowisko uruchomieniowe.</span><span class="sxs-lookup"><span data-stu-id="69f1e-124">Starting with the [!INCLUDE[net_v40_long](~/includes/net-v40-long-md.md)], <xref:System.AccessViolationException> exceptions thrown by the common language runtime are not handled by the `catch` statement in a structured exception handler if the exception occurs outside of the memory reserved by the common language runtime.</span></span> <span data-ttu-id="69f1e-125">Do obsługi takich <xref:System.AccessViolationException> wyjątek, należy zastosować <xref:System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptionsAttribute> atrybut do metody jest wyjątek.</span><span class="sxs-lookup"><span data-stu-id="69f1e-125">To handle such an <xref:System.AccessViolationException> exception, you should apply the  <xref:System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptionsAttribute> attribute to the method in which the exception is thrown.</span></span> <span data-ttu-id="69f1e-126">Ta zmiana nie wpływa na <xref:System.AccessViolationException> wyjątków zgłaszanych przez kod użytkownika, które mogą w dalszym ciągu podlegać `catch` instrukcji.</span><span class="sxs-lookup"><span data-stu-id="69f1e-126">This change does not affect <xref:System.AccessViolationException> exceptions thrown by user code, which can continue to be caught by a `catch` statement.</span></span> <span data-ttu-id="69f1e-127">Dla kodu napisanego dla wcześniejszych wersji programu .NET Framework, która ma zostać ponownie skompilować i uruchomić bez żadnych modyfikacji na [!INCLUDE[net_v40_long](~/includes/net-v40-long-md.md)], możesz dodać [ \<legacycorruptedstateexceptionspolicy — >](~/docs/framework/configure-apps/file-schema/runtime/legacycorruptedstateexceptionspolicy-element.md) elementu do użytkownika plik konfiguracji aplikacji.</span><span class="sxs-lookup"><span data-stu-id="69f1e-127">For code written for previous versions of the .NET Framework that you want to recompile and run without modification on the [!INCLUDE[net_v40_long](~/includes/net-v40-long-md.md)], you can add the [\<legacyCorruptedStateExceptionsPolicy>](~/docs/framework/configure-apps/file-schema/runtime/legacycorruptedstateexceptionspolicy-element.md) element to your app's configuration file.</span></span> <span data-ttu-id="69f1e-128">Należy pamiętać, że można również odbierać powiadomienia wyjątki Jeśli zdefiniowano funkcję obsługi <xref:System.AppDomain.FirstChanceException?displayProperty=nameWithType> lub <xref:System.AppDomain.UnhandledException?displayProperty=nameWithType> zdarzeń.</span><span class="sxs-lookup"><span data-stu-id="69f1e-128">Note that you can also receive notification of the exceptions if you have defined a handler for the <xref:System.AppDomain.FirstChanceException?displayProperty=nameWithType> or <xref:System.AppDomain.UnhandledException?displayProperty=nameWithType> event.</span></span>  
  
## <a name="version-information"></a><span data-ttu-id="69f1e-129">Informacje o wersji</span><span class="sxs-lookup"><span data-stu-id="69f1e-129">Version Information</span></span>  
 <span data-ttu-id="69f1e-130">Ten wyjątek w programie .NET Framework 2.0 i nowszych wersjach.</span><span class="sxs-lookup"><span data-stu-id="69f1e-130">This exception is thrown in the .NET Framework 2.0 and later versions.</span></span> <span data-ttu-id="69f1e-131">We wcześniejszych wersjach programu .NET Framework, naruszenia zasad dostępu w kodu niezarządzanego lub niebezpieczny kod zarządzany jest reprezentowana przez <xref:System.NullReferenceException> w kodzie zarządzanym.</span><span class="sxs-lookup"><span data-stu-id="69f1e-131">In earlier versions of the .NET Framework, an access violation in unmanaged code or unsafe managed code is represented by a <xref:System.NullReferenceException> in managed code.</span></span> <span data-ttu-id="69f1e-132">A <xref:System.NullReferenceException> jest również element zgłaszany, gdy odwołanie o wartości null jest wyłuskiwany w kodzie zarządzanym weryfikowalny, wystąpienie, które nie obejmują uszkodzenia danych, a nie istnieje sposób odróżnienie dwie sytuacje, w wersji 1.0 lub 1.1.</span><span class="sxs-lookup"><span data-stu-id="69f1e-132">A <xref:System.NullReferenceException> is also thrown when a null reference is dereferenced in verifiable managed code, an occurrence that does not involve data corruption, and there is no way to distinguish between the two situations in versions 1.0 or 1.1.</span></span>  
  
 <span data-ttu-id="69f1e-133">Administratorzy mogą umożliwić wybranych aplikacji powrócić do zachowania programu .NET Framework w wersji 1.1.</span><span class="sxs-lookup"><span data-stu-id="69f1e-133">Administrators can allow selected applications to revert to the behavior of the .NET Framework version 1.1.</span></span> <span data-ttu-id="69f1e-134">Umieść następujący wiersz w [ &lt;środowiska uruchomieniowego&gt; elementu](~/docs/framework/configure-apps/file-schema/runtime/runtime-element.md) sekcji pliku konfiguracji aplikacji:</span><span class="sxs-lookup"><span data-stu-id="69f1e-134">Place the following line in the [&lt;runtime&gt; Element](~/docs/framework/configure-apps/file-schema/runtime/runtime-element.md) section of the configuration file for the application:</span></span>  
  
```  
<legacyNullReferenceExceptionPolicy enabled = "1"/>  
```  
  
 ]]></format>
    </remarks>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="69f1e-135">Inicjuje nowe wystąpienie klasy <see cref="T:System.AccessViolationException" /> klasy.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-135">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AccessViolationException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AccessViolationException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AccessViolationException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="69f1e-136">Inicjuje nowe wystąpienie klasy <see cref="T:System.AccessViolationException" /> używając dostarczany przez system komunikatu opisującego błąd.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-136">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class with a system-supplied message that describes the error.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="69f1e-137">Ten konstruktor inicjuje <xref:System.Exception.Message%2A> właściwość nowego wystąpienia dostarczony przez system komunikat, który opisuje błąd, takich jak "podjęto próbę odczytu lub zapisu pamięci chronionej.</span><span class="sxs-lookup"><span data-stu-id="69f1e-137">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "Attempted to read or write protected memory.</span></span> <span data-ttu-id="69f1e-138">Jest to często wskazanie, że inna pamięć została uszkodzona."</span><span class="sxs-lookup"><span data-stu-id="69f1e-138">This is often an indication that other memory has been corrupted."</span></span> <span data-ttu-id="69f1e-139">Ten komunikat uwzględnia bieżącą kulturę systemu.</span><span class="sxs-lookup"><span data-stu-id="69f1e-139">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="69f1e-140">W poniższej tabeli przedstawiono początkowe wartości właściwości dla wystąpienia <xref:System.AccessViolationException>.</span><span class="sxs-lookup"><span data-stu-id="69f1e-140">The following table shows the initial property values for an instance of <xref:System.AccessViolationException>.</span></span>  
  
|<span data-ttu-id="69f1e-141">Właściwość</span><span class="sxs-lookup"><span data-stu-id="69f1e-141">Property</span></span>|<span data-ttu-id="69f1e-142">Wartość</span><span class="sxs-lookup"><span data-stu-id="69f1e-142">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="69f1e-143">`null`.</span><span class="sxs-lookup"><span data-stu-id="69f1e-143">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="69f1e-144">Zlokalizowany ciąg komunikatu o błędzie.</span><span class="sxs-lookup"><span data-stu-id="69f1e-144">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AccessViolationException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AccessViolationException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AccessViolationException(System::String ^ message);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="69f1e-145">Komunikat, który opisuje wyjątek.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-145">The message that describes the exception.</span>
          </span>
          <span data-ttu-id="69f1e-146">Obiekt wywołujący tego konstruktora jest wymagany w celu zagwarantowania, że ten ciąg został zlokalizowany w bieżącej kulturze systemu.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-146">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="69f1e-147">Inicjuje nowe wystąpienie klasy <see cref="T:System.AccessViolationException" /> klasy z określony komunikat zawierający opis błędu.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-147">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class with a specified message that describes the error.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="69f1e-148">W poniższej tabeli przedstawiono początkowe wartości właściwości dla wystąpienia <xref:System.AccessViolationException>.</span><span class="sxs-lookup"><span data-stu-id="69f1e-148">The following table shows the initial property values for an instance of <xref:System.AccessViolationException>.</span></span>  
  
|<span data-ttu-id="69f1e-149">Właściwość</span><span class="sxs-lookup"><span data-stu-id="69f1e-149">Property</span></span>|<span data-ttu-id="69f1e-150">Wartość</span><span class="sxs-lookup"><span data-stu-id="69f1e-150">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="69f1e-151">`null`.</span><span class="sxs-lookup"><span data-stu-id="69f1e-151">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="69f1e-152">Ciąg z komunikatem o określonych w `message`.</span><span class="sxs-lookup"><span data-stu-id="69f1e-152">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected AccessViolationException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AccessViolationException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; AccessViolationException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="69f1e-153">
            <see cref="T:System.Runtime.Serialization.SerializationInfo" /> Zawierający dane zserializowanego obiektu.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-153">The <see cref="T:System.Runtime.Serialization.SerializationInfo" /> that holds the serialized object data.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="69f1e-154">
            <see cref="T:System.Runtime.Serialization.StreamingContext" /> Zawierający informacje kontekstowe dotyczące źródła lub miejsca docelowego.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-154">The <see cref="T:System.Runtime.Serialization.StreamingContext" /> that contains contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="69f1e-155">Inicjuje nowe wystąpienie klasy <see cref="T:System.AccessViolationException" /> z zserializowanymi danymi.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-155">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class with serialized data.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="69f1e-156">Ten konstruktor jest wywoływana podczas deserializacji do odtworzenia obiektu wyjątku przesyłanego za pomocą strumienia.</span><span class="sxs-lookup"><span data-stu-id="69f1e-156">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="69f1e-157">Aby uzyskać więcej informacji, zobacz [XML i serializacji SOAP](~/docs/standard/serialization/xml-and-soap-serialization.md).</span><span class="sxs-lookup"><span data-stu-id="69f1e-157">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AccessViolationException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AccessViolationException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AccessViolationException(System::String ^ message, Exception ^ innerException);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="69f1e-158">Komunikat, który opisuje wyjątek.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-158">The message that describes the exception.</span>
          </span>
          <span data-ttu-id="69f1e-159">Obiekt wywołujący tego konstruktora jest wymagany w celu zagwarantowania, że ten ciąg został zlokalizowany w bieżącej kulturze systemu.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-159">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="69f1e-160">Wyjątek, który jest przyczyną bieżącego wyjątku.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-160">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="69f1e-161">Jeśli <c>innerException</c> parametr nie jest <see langword="null" />, bieżący wyjątek jest zgłaszany w <see langword="catch" /> bloku, który obsługuje wyjątek wewnętrzny.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-161">If the <c>innerException</c> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="69f1e-162">Inicjuje nowe wystąpienie klasy <see cref="T:System.AccessViolationException" /> z określonego komunikatu o błędzie i odwołania do wyjątek wewnętrzny będący przyczyną tego wyjątku.</span>
          <span class="sxs-lookup">
            <span data-stu-id="69f1e-162">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="69f1e-163">Wyjątek zgłaszany bezpośrednio w wyniku poprzedniego wyjątku powinien zawierać odwołanie do poprzedniego wyjątku we <xref:System.Exception.InnerException%2A> właściwości.</span><span class="sxs-lookup"><span data-stu-id="69f1e-163">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="69f1e-164"><xref:System.Exception.InnerException%2A> Właściwość zwraca tę samą wartość, która została przekazana do konstruktora, lub `null` Jeśli <xref:System.Exception.InnerException%2A> właściwości nie dostarcza do konstruktora wartości wyjątku wewnętrznego.</span><span class="sxs-lookup"><span data-stu-id="69f1e-164">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="69f1e-165">W poniższej tabeli przedstawiono początkowe wartości właściwości dla wystąpienia <xref:System.AccessViolationException>.</span><span class="sxs-lookup"><span data-stu-id="69f1e-165">The following table shows the initial property values for an instance of <xref:System.AccessViolationException>.</span></span>  
  
|<span data-ttu-id="69f1e-166">Właściwość</span><span class="sxs-lookup"><span data-stu-id="69f1e-166">Property</span></span>|<span data-ttu-id="69f1e-167">Wartość</span><span class="sxs-lookup"><span data-stu-id="69f1e-167">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="69f1e-168">`null`.</span><span class="sxs-lookup"><span data-stu-id="69f1e-168">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="69f1e-169">Ciąg z komunikatem o określonych w `message`.</span><span class="sxs-lookup"><span data-stu-id="69f1e-169">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>